icaruscode/PMT/OpReco/ICARUSFlashAssAna_module.cc

```diff
@@ -350,36 +537,97 @@ void opana::ICARUSFlashAssAna::analyze(art::Event const& e) {
   m_timestamp = e.time().timeHigh(); // precision to the second 
 
 
+  /*
+  This part is for the trigger information
+  */
+
+
   // We work out the trigger information here 
   if( !fTriggerLabel.empty() ) { 
 
+      // Beam information
       art::Handle<std::vector<sim::BeamGateInfo>> beamgate_handle;
       e.getByLabel( fTriggerLabel, beamgate_handle );
-
+      
       if( beamgate_handle.isValid() ) {
 
         for( auto const & beamgate : *beamgate_handle ) {
 
           m_beam_gate_start = beamgate.Start(); 
           m_beam_gate_width = beamgate.Width(); 
-          m_beam_type = beamgate.BeamType() ;
+          m_beam_type = beamgate.BeamType() ; 
 
         }
 
       }
 
       else {
-        std::cout << "Invalid Trigger Data product " << fTriggerLabel.label() << "\n" ;
+        if( fDebug) { std::cout << "No sim::BeamGateInfo associated to label: " << fTriggerLabel.label() << "\n" ; }
```

Comment:
I recommend either `MF_LOGTRACE()` or, if you really want to bypass message facility management of the debugging, `mf::LogVerbatim` (which outputs `INFO` verbosity level).